### DISCUSSION TITLE
[need_help] restoring singlebackup with restorechanbackup fails

### DISCUSSION THREAD

**Discussion by warioishere** [lightningnetwork/lnd#8478] [Troubleshooting]
### Background

I am trying to restore a single channel from a single backup file. It does not work, see details below.

### Your environment

* version of `lnd`
lnd 0.17.4 rc1 (containerized by btcpay-docker)
* which operating system (`uname -a` on *Nix)
Linux btcpay 6.5.11-7-pve SMP PREEMPT_DYNAMIC PMX 6.5.11-7 (2023-12-05T09:44Z) x86_64 x86_64 x86_64 GNU/Linux

* version of `btcd`, `bitcoind`, or other backend
26.0
* any other relevant environment details
Running the node on Proxmox

### Steps to reproduce

Trying to export a specific channel with `exportchanbackup --chan_point 9110b04f1ec66f20e3d7a8e0a8e15ce10f0a4c8729872c3a5cd036093829ef4b:1 --output_file=/root/.lnd/bitcoinsuisse.backup
`
and restoring with

`restorechanbackup --single_file=/root/.lnd/bitcoinsuisse.backup`

### Expected behaviour

channel will will be restored from single backup file

### Actual behaviour

Does not work:

`[lncli] unable to restore chan backups: rpc error: code = Unknown desc = unable to unpack single backups: could not derive private key for legacy channel revocation root format: unable to derive private key`

Additional Info: the restore was already triggered with a full restorechanbackup some months ago, but it did not trigger a forceclose on this specific channel on the other side propably due to high fee enviroment that time.

**Comment by guggero** [comment]
Are you exporting the backup from the node you initially ran `restorechanbackup` with the full file on? Do you still have the full file available?

**Comment by warioishere** [comment]
> Are you exporting the backup from the node you initially ran `restorechanbackup` with the full file on? Do you still have the full file available?

Hello Oli

yes its the same node,maybe you can remember the issue I had with my SCB case we have discussed arround Christmas/Newyear

I never changed or touched the original SCB file so I guess its the original SCB file, where I export the single backup now and from which I did the full restore that time ago.  I continued using the same node normally since then.

**PS: ah 'exportchanbackup' does not export the single backup from the original SCB  file I guess. Now I need a hint what to do because yes, this original file is still there because as I mentioned, never touched it.**

Greetings

**Comment by guggero** [comment]
Yes, I remember. And I think the issue is indeed that you're exporting a SCB from an SCB which is IMO not fully supported.
What you can do instead is take the original file and use [`chantools filterbackup`](https://github.com/lightninglabs/chantools/blob/master/doc/chantools_filterbackup.md) to reduce the file to one that only has the channel in question (meaning filter out all other channels). Then use that (as a `multi_file` but with just one entry).

**Comment by guggero** [comment]
But I'm not sure doing so would change anything. If you say the node you used the full SCB on is still running like that, then it should periodically try and request the peer to force close. Or at least that's what I think it should. Maybe it only happens on startup though. So probably still worth a try, just want to manage expectations.

**Comment by warioishere** [comment]
> Yes, I remember. And I think the issue is indeed that you're exporting a SCB from an SCB which is IMO not fully supported. What you can do instead is take the original file and use [`chantools filterbackup`](https://github.com/lightninglabs/chantools/blob/master/doc/chantools_filterbackup.md) to reduce the file to one that only has the channel in question (meaning filter out all other channels). Then use that (as a `multi_file` but with just one entry).

I understand the issue when you told me that I exported a channelback from an already restored scb channel. 
I will try chantools later. 

Do you think chantools triggerforce might do the trick?

I have contact with bitcoinsuisse, they cannot manually force close, they say they have the issue that says `channel already closed`  when they try to force close again. 
I dont get much information because I talk to a relation manager which passed me few infos from the IT department. I told them already to raise mempool, restart their node afterwards and do this and that. But I get an answer with almost no Infos once in a week and I dont know what they already tried and what they didnt try yet. Just saying infrastrural problems.

**Reply by guggero** [reply]
Then they need to look at their local channel list, find out the closing transaction and re-broadcast it. Or restart their node.

> Do you think chantools triggerforce might do the trick?

No, unless they're running CLN on the specific version the command was created for.

**Reply by warioishere** [reply]
I guess they did that, at least to restart the node. So actually, theres no need to retriggert SCB via restorechanbackup? I thought about doing so because fees are low now. I of course restarted my node several times, reconnect to bitcoinsuisse and saw the the DLP action in lnd.logs the after restarting the node. 
I am writing them since start of Jan and I dont see any closing from their side. What other options do I have to get back my 2mil sats if they dont cooperate?

**Reply by guggero** [reply]
Yes, I think the DLP trigger is already happening, so not sure you can do anything else on your side... Probably a problem in their setup. I would try to insist them attempting to re-broadcast the closing TX (if we knew what node implementation and version they're running, I could suggest commands to run...)

**Reply by warioishere** [reply]
We checked their feature bits on their node and we guess that they are running LND, but not sure what version

This is what I told them to do so far: 

- increase their mempool during the full mempool some weaks ago.
- do a `lncli closechannel --force --chan_point 9110b04f1ec66f20e3d7a8e0a8e15ce10f0a4c8729872c3a5cd036093829ef4b:1 `
>>they said something like we are getting a `channel already closed` from that

- check their `lncli pendingchannels` and rebroadcast localclosing tx
- restart their nodes will do the trick without doing anything


Thats what I told them so far. Any other hints? 

ty

**Reply by guggero** [reply]
Yes, that sounds good. If they have trouble finding the "localclosing tx", it _should_ appear in `lncli listchaintxns` with the full hex so they can re-broadcast manually.

**Reply by warioishere** [reply]
I will insist again ask them to send me the output of their  `lncli pendingchannels`
and their localclosingtx as a hex from `lncli listchaintxns --start_height -1 --end_height <curent_block_height>`

and see what get, then I will try to tell them to republish with `lncli wallet publishtx <closinghx>`

I dont know what to believe, if they dont know how to operate their node, or if they really suffer from a bug.

I'll let you know if you are interessted.

Greetings

**Reply by warioishere** [reply]
Hi @guggero 

Bitcoinsuisse finally made it to close this channel, guess they had no time or it had no priority. The closing is already confirmed:

https://mempool.space/tx/6d291145ee041279563f4ea94985df758775c3d66f51032c488e8cb0bb56ece0

However, the sats are not on my wallet. `lncli wallet pendingsweeps` gives me:

```
root@btcpay:~/btcpayserver-docker/docker-compose-generator/docker-fragments# bitcoin-lncli.sh wallet pendingsweeps
{
        "pending_sweeps": [
                {
                        "outpoint": "6d291145ee041279563f4ea94985df758775c3d66f51032c488e8cb0bb56ece0:1",
                        "witness_type": "COMMITMENT_TO_REMOTE_CONFIRMED",
                        "amount_sat": 1996530,
                        "sat_per_vbyte": 0,
                        "broadcast_attempts": 0,
                        "next_broadcast_height": 831428,
                        "requested_sat_per_vbyte": 0,
                        "requested_conf_target": 6,
                        "force": false
                },
                {
                        "outpoint": "6d291145ee041279563f4ea94985df758775c3d66f51032c488e8cb0bb56ece0:0",
                        "witness_type": "COMMITMENT_ANCHOR",
                        "amount_sat": 330,
                        "sat_per_vbyte": 1,
                        "broadcast_attempts": 0,
                        "next_broadcast_height": 831428,
                        "requested_sat_per_vbyte": 1,
                        "requested_conf_target": 0,
                        "force": false
                }
        ]
}

```

I already restarted lnd. I have         `sweeper.batchwindowduration=2h`       and this time already passed by. 
I find ` "requested_sat_per_vbyte": 0,` and  `"requested_sat_per_vbyte": 1,` for the anchor pretty weird. 

I dont have any txns in listchaintxns

This is the output of `lncli pendingchannels`

```
"pending_force_closing_channels": [
        {
            "channel": {
                "remote_node_pub": "027222caea9c1070952bc8c7a05a6f75f26f3bffc37e80e627cc651d50d50eb268",
                "channel_point": "9110b04f1ec66f20e3d7a8e0a8e15ce10f0a4c8729872c3a5cd036093829ef4b:1",
                "capacity": "2000000",
                "local_balance": "1996530",
                "remote_balance": "0",
                "local_chan_reserve_sat": "0",
                "remote_chan_reserve_sat": "0",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "0",
                "chan_status_flags": "",
                "private": true,
                "memo": ""
            },
            "closing_txid": "6d291145ee041279563f4ea94985df758775c3d66f51032c488e8cb0bb56ece0",
            "limbo_balance": "1996860",
            "maturity_height": 831423,
            "blocks_til_maturity": -8,
            "recovered_balance": "0",
            "pending_htlcs": [],
            "anchor": "LIMBO"
        }
    ],
```


blocks till maturiy is already negativ.

You got a hint for me what to do? Just wait?

Thankyou very much 

Wario

**Reply by guggero** [reply]
Looks like the sweep just confirmed: https://mempool.space/tx/56258c0024fd0260d409317702cf709e4d7dd0b8e55dcff9ca6c66285cebcbbe

**Reply by warioishere** [reply]
yes saw it now, I didnt know how those sweeps and stuff worked. Learned a bit more by reading
https://docs.lightning.engineering/the-lightning-network/payment-channels/understanding-sweeping

so this topic can be finally put at acta :)
thanks again for you time and your help!

Have a nice evening Oli!
